src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -298,6 +293,41 @@ void HierRTLMP::hierRTLMacroPlacer()
                 util,
                 core_util);
 
+  //
+  // Set defaults for min/max number of instances and macros if not set by user.
+  //
+  if (max_num_macro_base_ <= 0 || min_num_macro_base_ <= 0
+      || max_num_inst_base_ <= 0 || min_num_inst_base_ <= 0) {
+    min_num_inst_base_ = std::floor(
+        metric_->getNumStdCell() / std::pow(coarsening_ratio_, max_num_level_));
+    if (min_num_inst_base_ <= 1000)
+      min_num_inst_base_ = 1000;  // lower bound
+    max_num_inst_base_ = min_num_inst_base_ * coarsening_ratio_ / 2.0;
+    min_num_macro_base_ = std::floor(
+        metric_->getNumMacro() / std::pow(coarsening_ratio_, max_num_level_));
+    if (min_num_macro_base_ <= 0)
+      min_num_macro_base_ = 1;  // lowerbound
+    max_num_macro_base_ = min_num_macro_base_ * coarsening_ratio_ / 2.0;
+
+    if (metric_->getNumMacro() <= 150) {
+      max_num_level_
+          = 1;  // if the number of macros is small we can do single level
+    }
+  }
+
+  unsigned coarsening_factor = std::pow(coarsening_ratio_, max_num_level_ - 1);
+  max_num_macro_base_ = max_num_macro_base_ * coarsening_factor;
+  min_num_macro_base_ = min_num_macro_base_ * coarsening_factor;
+  max_num_inst_base_ = max_num_inst_base_ * coarsening_factor;
+  min_num_inst_base_ = min_num_inst_base_ * coarsening_factor;
```

Comment:
Should these *_base_ be local variables rather than class members?  I don't see them used elsewhere.

Reply:
they are used during clustering.